src/DotNet.RateLimiter/ActionFilters/RateLimitAttribute.cs

```diff
@@ -159,20 +160,14 @@ private string ProvideRateLimitKey(ActionExecutingContext context, string reques
             }
             if (!string.IsNullOrWhiteSpace(BodyParams))
             {
-                var parameters = BodyParams.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
                 var jsonSerializer = JsonConvert.SerializeObject(context.ActionArguments);
                 var obj = JObject.Parse(jsonSerializer);
-                var properties = obj.Descendants().OfType<JProperty>().Where(p => p.Value.Type != JTokenType.Array && p.Value.Type != JTokenType.Object).ToList();
-                foreach (var parameter in parameters)
+
+                if (obj.Root.First().Values().FirstOrDefault(x => x.Type == JTokenType.Property && string.Equals(((JProperty)x).Name, BodyParams, StringComparison.OrdinalIgnoreCase)) is JProperty property)
```

Comment:
For better code readability I think this one is better (of course you need to check it with splitted bodyparams):
```csharp
var rootProperty = obj.Root.First().Values().FirstOrDefault(x => x.Type == JTokenType.Property
                            && string.Equals(((JProperty)x).Name, BodyParams, StringComparison.OrdinalIgnoreCase));

                if (rootProperty is JProperty property)
                {
                    rateLimitKey.Append(property.Value).Append(":");
                }
```